tiledb/sm/misc/uri.cc

```diff
@@ -148,7 +148,10 @@ std::string URI::last_path_part() const {
 }
 
 URI URI::parent() const {
-  uint64_t pos = uri_.find_last_of('/');
+  auto uri = uri_;
```

Comment:
@tdenniston any objections to this change here? It was apparently causing problems when `uri_` was ending in `/`.

Reply:
I think you should add a check for empty strings, e.g. `if (uri_.empty()) return URI();`, as calling `.back()` on an empty string is undefined.

Reply:
Good point, thanks.